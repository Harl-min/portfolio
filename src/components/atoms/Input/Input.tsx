import classNames from 'classnames';

export interface InputProps {
	label: string;
	id: string;
	isRequired?: boolean;
	error?: string;
	[rest: string]: unknown;
}

export const Input = ({
	label,
	id,
	isRequired,
	error,
	...rest
}: InputProps) => {
	return (
		<div className="flex flex-col gap-1">
			<label htmlFor={id} className="font-bold">
				{label} {isRequired && <span className="text-red-700">*</span>}
			</label>
			<input
				id={id}
				className={classNames(
					'my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent',
					{
						'ring-red-600': !!error,
						'ring-grey-400 dark:ring-slate-500': !error,
					}
				)}
				{...rest}
			/>
			{error && (
				<div className="font-sm font-normal -mt-2 text-red-600">{error}</div>
			)}
		</div>
	);
};
