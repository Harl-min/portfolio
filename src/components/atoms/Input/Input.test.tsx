import { Input } from '../Input';

import { render } from '@testing-library/react';

const mockField = {
	value: '',
	checked: false,
	onChange: jest.fn(),
	onBlur: jest.fn(),
	multiple: undefined,
	name: 'password',
};
const mockMeta = {
	touched: false,
	error: '',
	initialError: '',
	initialTouched: false,
	initialValue: '',
	value: '',
};

describe('Input', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Input
				label="Some input"
				id="some-input"
				placeholder="Some placeholder"
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
