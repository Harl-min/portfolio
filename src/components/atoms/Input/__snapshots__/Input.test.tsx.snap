// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-1"
  >
    <label
      class="font-bold"
      for="some-input"
    >
      Some input
       
    </label>
    <input
      class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
      id="some-input"
      placeholder="Some placeholder"
    />
  </div>
</div>
`;
